pkg/ndctl/region.go

```diff
@@ -208,13 +204,25 @@ func (r *region) CreateNamespace(opts CreateNamespaceOpts) (Namespace, error) {
 	}
 
 	if opts.Size != 0 {
-		ways := uint64(C.ndctl_region_get_interleave_ways(r))
-		align = align * ways
-		if opts.Size%align != 0 {
-			// Round up size to align with next block boundary.
-			opts.Size = (opts.Size/align + 1) * align
-			klog.V(4).Infof("%s: namespace size must align to interleave-width:%d * alignment:%d, force-align to %d",
-				regionName, ways, align, opts.Size)
+		namespacealign := align * r.InterleaveWays()
```

Comment:
What if `Size` really is zero?

Shouldn't we then create a minimal-sized namespace, i.e. go through this alignment code and round up?

Reply:
From LVM dev-mgr, we do not call with zero arg. There has been check for too small/zero value in previous code. Now in simplifying stage, I tried to remove it, and I saw one test (namespace conversion IIRC) caused call of LVM-mode creation with zero size, which did not went OK, so I restored check there and we only call Create if size >0.
In direct case, we always use size what is passed into CreateDevice without checking, and IIRC thats always been the case and has not caused trouble. I believe libndctl then creates max-sized namespace, at least that happens in cmd-line.
yes your idea is good, lets check it and create minimal i.e. lcmalign-sized

Reply:
> In direct case, we always use size what is passed into CreateDevice without checking, and IIRC thats always been the case and has not caused trouble.

"not caused trouble" is a dangerous argument - it could be that no-one has ever tested that...

So we agree, let's treat "zero size" as "smallest possible namespace".